mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -369,6 +369,85 @@ void RockDialect::initialize() {
 //===----------------------------------------------------------------------===//
 // Convolution operations
 //===----------------------------------------------------------------------===//
+static ConvolutionDims obtainConvDims(Operation *op) {
+  auto filterLayoutAttr = op->getAttrOfType<ArrayAttr>("filter_layout");
+  auto inputLayoutAttr = op->getAttrOfType<ArrayAttr>("input_layout");
+  auto outputLayoutAttr =
+      op->template getAttrOfType<ArrayAttr>("output_layout");
+
+  // Get shape of filter tensor.
+  auto filterType = op->getOperand(0).getType().template cast<MemRefType>();
+  ArrayRef<int64_t> filterShape = filterType.getShape();
+
+  // Get shape of input tensor.
+  auto inputType = op->getOperand(1).getType().template cast<MemRefType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+
+  // Get shape of output tensor.
+  auto outputType = op->getOperand(2).getType().template cast<MemRefType>();
+  ArrayRef<int64_t> outputShape = outputType.getShape();
+
+  int64_t y, x, ho, wo, hi, wi, k, c, n, g;
+  y = x = ho = wo = hi = wi = k = c = n = g = 0;
+
+  for (unsigned i = 0; i < filterLayoutAttr.size(); ++i) {
+    auto filterAttr = filterLayoutAttr.getValue()[i].cast<StringAttr>();
+    auto inputAttr = inputLayoutAttr.getValue()[i].cast<StringAttr>();
+    auto outputAttr = outputLayoutAttr.getValue()[i].cast<StringAttr>();
+
+    if (filterAttr.getValue() == "y") {
+      y = filterShape[i];
+    } else if (filterAttr.getValue() == "x") {
+      x = filterShape[i];
+    } else if (filterAttr.getValue() == "k") {
+      k = filterShape[i];
+    } else if (filterAttr.getValue() == "c") {
+      c = filterShape[i];
+    } else if (filterAttr.getValue() == "g") {
+      g = filterShape[i];
+    }
+
+    if (inputAttr.getValue() == "hi") {
```

Comment:
also switch

Reply:
Can you switch/case on a string?
And I'm not sure StringSwitch will work because we've got variable assignments in here.

(I was copying existing code from `AffixTuningParams`)

Reply:
Just the leading char.

Reply:
I'd say it's reasonable to check the exact value here